model/model_builder.py

```diff
@@ -178,6 +183,49 @@ def add_output_layer(self):
 
         return self
 
+    def add_secondary_output(self):
```

Comment:
This is very similar to the other output function. Is it possible to combine them and maybe take the labels as an input or something?